src/index.ts

```diff
@@ -0,0 +1,51 @@
+// =======================
+// Domain: Event publish/subscribe
+// =======================
+
+import sbp from '@sbp/sbp'
+import '@sbp/okturtles.data'
+
+const listenKey = (evt: string) => `events/${evt}/listeners`
+
+type ThisType = {
+  errorHandler?: (event: string, e: unknown) => void
+}
+
+export default (sbp('sbp/selectors/register', {
+  'okTurtles.events/_init': function (this: ThisType) {
+    this.errorHandler = (event: string, e: unknown) => {
+      console.error(`[okTurtles.events] Error at handler for ${event}`, e)
+    }
+  },
+  'okTurtles.events/on': function (event: string, handler: (...args: unknown[]) => void) {
+    sbp('okTurtles.data/add', listenKey(event), handler)
+    return () => sbp('okTurtles.events/off', event, handler)
+  },
+  'okTurtles.events/once': function (event: string, handler: (...args: unknown[]) => void) {
+    const cbWithOff = (...args: unknown[]) => {
+      handler(...args)
+      sbp('okTurtles.events/off', event, cbWithOff)
+    }
+    return sbp('okTurtles.events/on', event, cbWithOff)
+  },
+  'okTurtles.events/emit': function (this: ThisType, event: string, ...data: unknown[]) {
```

Comment:
We can't change the selector's function signature. This needs to accept 2 arguments as before, with no `this`

Also, SBP automatically assigns a `this` - you don't need to pass it in.

Reply:
This doesn't change the signature